### ISSUE TITLE
No preview available

### ISSUE THREAD

**Issue by enricodeleo** [vernaillen/wpnuxt-module#62]
Trying to render the preview page (/preview) I cannot see any content

![image](https://github.com/vernaillen/wpnuxt-module/assets/3534555/b9a9d169-e778-4e82-95e5-05c77f318e5e)

I created this pages in the `pages` directory

```vue
<template>
  <div class="prose dark:prose-invert">
    <div class="flex gap-4 items-center">
      <h3 class="m-0">Preview</h3>
      <NuxtLink :to="postEditUrl">edit</NuxtLink>
    </div>

    <hr />

    <client-only>
      <BlockRenderer v-if="page" :blocks="page?.editorBlocks" />
      <div v-else>Oops, page === null</div>
    </client-only>
  </div>
</template>

<script setup>
const route = useRoute()
const { preview_id: previewId } = route.query

const postEditUrl = useWPUri().postEdit(previewId)
const page = ref()

if (previewId) {
  const id = Number.parseInt(previewId.toString())
  try {
    const data = await usePageById(id, true)
    page.value = data?.data
  } catch (error) {
    console.error(error)
  }
}
</script>

```

Apparently the result from data is always `null`. I can't understand what's wrong. The id is absolutely correct. The url, coming from the wordpress editor when I click preview is like `/preview?preview_id=92&code=YyFeTrDXiOG5cGZo29sRZWI49DwbRk1mmZa2Z8QR7rs5S2%2Bl549A50fA5bauLgEOd%2BSLtweWkwUac1CkZjD/Ag==`. Any ideas?

**Comment by vernaillen** [comment]
Hi Enrico,
I will try to reproduce this, but probably won't have time any more today.
Yesterday I made a lot of changes to add serverside caching, and haven't retested everything yet. I will do that very soon.

**Comment by enricodeleo** [comment]
Thanks @vernaillen! It's free software, nobody can ask for same-day help :D I appreciated. I tried to understand on my own if something was wrong but the composable looks good I can't catch what's wrong. Maybe we should use the `code` part too ? 
If it helps it wasn't working for me even with the previous versions.

**Comment by vernaillen** [comment]
Hi @enricodeleo 
It seems the authorization header was missing to fetch the preview data. It's something I had missed when doing the rework for the serverside caching.   
That's now fixed in the latest release [v0.3.4](https://github.com/vernaillen/wpnuxt-module/releases/tag/v0.3.4)
